libgoal/libgoal.go

```diff
@@ -948,10 +975,48 @@ func (c *Client) GetPendingTransactions(maxTxns uint64) (resp v1.PendingTransact
 
 // GetPendingTransactionsByAddress gets a snapshot of current pending transactions on the node for the given address.
 // If maxTxns = 0, fetches as many transactions as possible.
-func (c *Client) GetPendingTransactionsByAddress(addr string, maxTxns uint64) (resp v1.PendingTransactions, err error) {
+func (c *Client) GetPendingTransactionsByAddress(addr string, maxTxns uint64) (resp generatedV2.PendingTransactionsResponse, err error) {
 	algod, err := c.ensureAlgodClient()
 	if err == nil {
-		resp, err = algod.PendingTransactionsByAddr(addr, maxTxns)
+		resp, err = algod.PendingTransactionsByAddrV2(addr, maxTxns)
+	}
+	return
+}
+
+// PendingTransactionsStruct represents a parsed PendingTransactionsResponse struct.
+type PendingTransactionsStruct struct {
```

Comment:
@algochoi Optional - Unless there's a rationale I missed, I think we can omit the `Struct` suffix.  As a minor thought, can the suffix be dropped?

If you're up for the change, I opened a child PR:  https://github.com/algochoi/go-algorand/pull/25.  Otherwise, feel welcomed to resolve as is.

Reply:
LGTM, thanks!